cmake/GenerateDateTimeFile.cmake

```diff
@@ -0,0 +1,10 @@
+# Creates "bout++-time.cxx" in the build directory with the
+# compilation date and time as variables
+
+string(TIMESTAMP bout_date "%b %d %Y")
+string(TIMESTAMP bout_time "%H:%M:%S")
+
+set(bout_date_time_file
+  "const char* boutcompiledate{\"${bout_date}\"}; const char* boutcompiletime{\"${bout_time}\"};")
```

Comment:
Why not use the compiler macros `__DATE__` and `__TIME__`?
If it is purely so that the file changes, adding a comment with "generated at" should do the job?

Reply:
Good shout, using the macros makes it simpler. This file is always re-generated if any other file changes, so the macros are sufficient